observations/forward_operators/obs_def_cice_mod.f90

```diff
@@ -386,10 +493,43 @@ subroutine get_expected_agreg_thickness(state_handle, ens_size, location, &
 where(istatus1 /= 0) istatus = istatus1
 where(istatus2 /= 0) istatus = istatus2
 !where(istatus == 0.and.agreg_sic>1e-6) agreg_thickness = agreg_volume/agreg_sic
-where(istatus  == 0) agreg_thickness = agreg_volume/max(agreg_sic,1e-8)
+where(istatus  == 0) agreg_thickness = max(agreg_volume,0.0_r8)/max(agreg_sic,1e-6_r8)
 
 end subroutine get_expected_agreg_thickness
 
+!-----------------------------------------------------------------------------
+!> For thickness, it's sum(volume*concentr)
+!> Good for sea ice thickness and snow thickness
+
+subroutine get_expected_agreg_concentration(state_handle, ens_size, location, &
+               obstype, agreg_sic, istatus)
+
+type(ensemble_type), intent(in)  :: state_handle
+integer,             intent(in)  :: ens_size
+type(location_type), intent(in)  :: location
+integer,             intent(in)  :: obstype
+real(r8),            intent(out) :: agreg_sic(ens_size)
+integer,             intent(out) :: istatus(ens_size)
+
+integer   :: istatus1(ens_size), istatus2(ens_size)
```

Comment:
no need for istatus1, istatus2 in this  routine, can just use istatus directly since you only have one call that returns an istatus.